src/components/header/header-with-search/HeaderWithSearch.tsx

```diff
@@ -0,0 +1,51 @@
+import "./HeaderWithSearch.css";
+import React, {useState} from "react";
+import SearchForm from "../search-form/SearchForm";
+import MovieDialog from "../../dialogs/movie-dialog/MovieDialog";
+import {Movie, MovieDialogState} from "../../../models/movies";
+
+function HeaderWithSearch(): React.JSX.Element {
+    const [movieDialogData, setMovieDialogData] = useState<MovieDialogState>({
+        isOpen: false,
+        title: '',
+        movie: null,
+    });
```

Comment:
Same here. No need for a complex state object. Just save the movie an it should be enough.